### PR TITLE
feat: Make service work with json (DEV-3930)

### PR DESCRIPTION
**Pull request by seakayone**
- **Add filter and query params to projects endpoint**
- **Add paging to projects endpoint**
- **Serve data from json files**
- **Fix loading assets in webapp in subpages**
- **Introduce project.status to json schemas**
- **Remove `_*.json` data**

Basically you can ignore the changes of the `*.json` files. I have copied over the latest version from the Go service and added the `project.status` property with the python script from this pr. That also reformatted all json files, hence so many changed lines.

### CODE REVIEW

File: docs/domain_model/schema-metadata.json [hunk 1]
```diff
@@ -159,6 +159,15 @@
...
           "pattern": "^[0-9A-F]{4}$"
         },
+        "status": {
+          "type": "string",
+          "description": "The status of the dataset.",
+          "enum": [
+            "ongoing",
+            "finished"
+          ],
+          "default": "Ongoing"
```
Comment (seakayone) [inline docs/domain_model/schema-metadata.json:169]:
```suggestion
          "default": "ongoing"
```

File: dsp-meta/resources/schema-metadata-draft.json [hunk 2]
```diff
@@ -159,6 +159,15 @@
...
           "pattern": "^[0-9A-F]{4}$"
         },
+        "status": {
+          "type": "string",
+          "description": "The status of the dataset.",
+          "enum": [
+            "ongoing",
+            "finished"
+          ],
+          "default": "Ongoing"
```
Comment (seakayone) [inline dsp-meta/resources/schema-metadata-draft.json:169]:
```suggestion
          "default": "ongoing"
```

File: data/add_status.python
Comment (BalduinLandolt) [inline data/add_status.python:1]: question: why `.python`, maybe I don't get the reason behind it, but I would have expected it to be `.py`

File: data/add_status.python [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		"083E",
+		"0843",
+		"0844",
+		"0849",
+		"084A",
+		"084E"
+]
+
+def update_json_files(folder_path):
+    print (ongoing_shortcodes)
```
Comment (BalduinLandolt) [inline data/add_status.python:47]: nitpick (this is not about python, I know): this looks a lot like Python 2 to me. In modern Python, `print` is a function like any other, so one should not include a space before parenthesis.

File: dsp-meta-cmd/src/main-server.rs [hunk 4]
```diff
@@ -46,8 +46,6 @@ fn main() {
 }
 
 async fn init_server() {
-    info!("Ivan was here!");
-
```
Comment (BalduinLandolt) [inline dsp-meta-cmd/src/main-server.rs:50]: comment (jocular): I was hoping to be the one removing this line. I would have replaced it with `info!("Removing graffiti... done.");` 😄

File: dsp-meta/src/api/router.rs [hunk 5]
```diff
@@ -18,33 +19,32 @@ use crate::app_state::AppState;
...
 pub fn router(shared_state: Arc<AppState>) -> Router {
+    let cors = CorsLayer::new()
+        .expose_headers(Any)
+        .allow_methods(Any)
+        .allow_headers(Any)
+        .allow_origin(Any);
     Router::new()
         .route(
-            "/api/projects",
+            "/api/v1/projects",
```
Comment (BalduinLandolt) [inline dsp-meta/src/api/router.rs:29]: praise: great! I'm in favour of adding versioning to all public APIs

**Review by BalduinLandolt — APPROVED** [review]
Great job!

File: data/add_status.python
Comment (seakayone) [reply data/add_status.python:1]: => https://github.com/dasch-swiss/dsp-meta/pull/146/commits/81ca39f7b79773952afc09b93b42a95da60aa6b3

File: data/add_status.python (hunk 3 above)
Comment (seakayone) [reply data/add_status.python]: => https://github.com/dasch-swiss/dsp-meta/pull/146/commits/81ca39f7b79773952afc09b93b42a95da60aa6b3

File: dsp-meta/src/api/router.rs (hunk 5 above)
Comment (seakayone) [reply dsp-meta/src/api/router.rs:29]: Thanks, but no praises for me please. This was the API in the go service, I have simply aligned the code to be compatible with the existing frontend.

File: dsp-meta-cmd/src/main-server.rs (hunk 4 above)
Comment (subotic) [reply dsp-meta-cmd/src/main-server.rs:50]: hmm, don't know if I like this ...

File: web-frontend/src/project-page/ProjectWidget.svelte [hunk 6]
```diff
@@ -134,45 +168,53 @@
...
-      {#each $projectMetadata?.project.funders.map((o) => {return findObjectByID(o)}) as f}
-          {#if f.__type === "Person"}
-            <!-- TODO: handle funding person - need to find example -->
-            <!-- <div class=data>{findObjectById(f)?.givenName.split(";").join(" ")} {findObjectById(f)?.familyName}</div> -->
-          {:else if f.__type === "Organization"}
-            <div class=data>{f.name}</div>
-          {/if}
+      {#each $projectMetadata?.project.funders.map((o) => {
+        return findObjectByID(o)
+      }) as f}
```
Comment (mpro7) [inline web-frontend/src/project-page/ProjectWidget.svelte:189]: Is this done automatically by some kind of formatter? For me this makes entire HTML a bit less readable. I know that some of the lines containing functions are long, but it is either only my feeling, or just entire structure of SVELTE way of HTML code with many statements. My just 2 rappens for later discussion, not something to worry now.

File: web-frontend/src/project-page/ProjectWidget.svelte [hunk 7]
```diff
@@ -209,64 +251,73 @@
...
-    margin: 15px 0;
-    padding: 0 10px 10px;
-    box-shadow: var(--shadow-1);
-  }
-  .chevron {
-    padding: 0;
-    height: 100%;
-    vertical-align: text-bottom;
-  }
+    a {
```
Comment (mpro7) [inline web-frontend/src/project-page/ProjectWidget.svelte:268]: Another formatting detail, entire file had 2 spaces indention, IMO it should remain like that.

File: web-frontend/src/store.ts [hunk 8]
```diff
@@ -1,52 +1,62 @@
...
-  })
+  });
 
-  console.log(baseUrl, route);
   navigate(`/${route}`);
 
-  await fetch(`${baseUrl}api/${route}`)
+  await fetch(`${(baseUrl())}/api/v1/${route}`)
     .then(r => {
+      console.log("RESPONSE count header:", r.headers.get('X-Total-Count'));
```
Comment (mpro7) [inline web-frontend/src/store.ts:49]: Is this log left for purpose?

File: web-frontend/src/store.ts [hunk 9]
```diff
@@ -1,52 +1,62 @@
...
       const totalCount = parseInt(r.headers.get('X-Total-Count'));
       let totalPages = Math.floor(totalCount / baseResultsRange[1]);
       if (!Number.isInteger(totalCount / baseResultsRange[1])) {
         totalPages++;
-      };
-      pagination.set({currentPage: page, currentResultsRange, totalCount, totalPages});
+      }
+      ;
```
Comment (mpro7) [inline web-frontend/src/store.ts:55]: Weird formatting.

File: web-frontend/src/project-page/ProjectWidget.svelte (hunk 6 above)
Comment (seakayone) [reply web-frontend/src/project-page/ProjectWidget.svelte:189]: I like that maybe you create a followup issue to improve the svelte setup. I would like that.

File: web-frontend/src/project-page/ProjectWidget.svelte (hunk 7 above)
Comment (seakayone) [reply web-frontend/src/project-page/ProjectWidget.svelte:268]: I like that maybe you create a followup issue to improve the svelte setup. I would like that.

File: web-frontend/src/project-page/ProjectWidget.svelte [hunk 10]
```diff
@@ -12,41 +19,64 @@
...
-    {/if}
-    <!-- Secondary URL -->
-    {#if $projectMetadata?.project.secondaryURL}
-      <a class="data" href={$projectMetadata?.project.secondaryURL.url} target=_>
-        {truncateString($projectMetadata?.project.secondaryURL.text)}
-        <img class=chevron src="assets/icon/Chevron_right.svg" alt="chevron right indicating a link" />
-      </a>
-    {/if}
-  </div>
+  {#if ($projectMetadata?.project.url || $projectMetadata?.project.secondaryURL) && !$isTestEnvironment}
```
[the remark this answers is not in the document]
Comment (seakayone) [reply web-frontend/src/project-page/ProjectWidget.svelte:29]: You have an idea?

File: web-frontend/src/store.ts (hunk 9 above)
Comment (seakayone) [reply web-frontend/src/store.ts]: Yes => a263a60e5ae35aaa846869cb1870a7b757a2ea40

File: web-frontend/src/store.ts (hunk 8 above)
Comment (seakayone) [reply web-frontend/src/store.ts]: No => a263a60e5ae35aaa846869cb1870a7b757a2ea40

File: web-frontend/src/project-page/ProjectWidget.svelte (hunk 6 above)
Comment (mpro7) [reply web-frontend/src/project-page/ProjectWidget.svelte:189]: That is good idea. Could you precize what do you like then? Current formatting or previous approach (less line breaks)?

File: web-frontend/src/project-page/ProjectWidget.svelte (hunk 10 above)
[the remark this answers is not in the document]
Comment (mpro7) [reply web-frontend/src/project-page/ProjectWidget.svelte:29]: NOt yet, but if I will have, will open a PR.

**Pull request merged**